src/helper/bitmap.js

```diff
@@ -275,27 +275,79 @@ const floodFillInternal_ = function (x, y, imageData, newColor, oldColor, stack)
     }
 };
 
+/**
+ * Given a fill style string, get the color
+ * @param {string} fillStyleString the fill style
+ * @return {Array<int>} Color, a length 4 array
+ */
+const fillStyleToColor_ = function (fillStyleString) {
+    const tmpCanvas = document.createElement('canvas');
+    tmpCanvas.width = 1;
+    tmpCanvas.height = 1;
+    const context = tmpCanvas.getContext('2d');
+    context.fillStyle = fillStyleString;
+    context.fillRect(0, 0, 1, 1);
+    return context.getImageData(0, 0, 1, 1).data;
```

Comment:
there are not enough emoji to contain how clever/bewildering this is. I mean, it works... 🔥